domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -57,12 +57,14 @@ import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.launch
 import java.util.Date
 import javax.inject.Inject
+import javax.inject.Singleton
 
 private const val TAG = "CameraXCameraUseCase"
 
 /**
  * CameraX based implementation for [CameraUseCase]
  */
+@Singleton
```

Comment:
I changed this class to singleton scope so that it can be shared in multiple view models. I feel like CameraX lifecycle is intended to be application scoped more than viewmodel/activity scoped, specially considering how ProcessCameraProvider itself is a singleton.

But I wonder if this aligns with the current dependency injection scoping intentions for JCA? If not, need to think about workarounds for this.

Reply:
This sounds reasonable to me, especially since we are using `ProcessCameraProvider`. 

I would like to think through some of the implications, however. Are there any cases where this could cause state to persist when we don't expect it to- mainly because this class has instance variables that wouldn't be cleared if the class instance is sticking around. I'm thinking about the case where we transition to a gallery or some part of the app that doesn't use camera. Maybe this means we would need to be extra diligent about always calling `initialize()` when we need to reset state`?

Reply:
Yeah I think it should be integrated into `PreviewViewModel`. While technically `ScreenFlash` could be an independent screen, it doesn't make much sense in isolation, except for maybe a few pathological cases. Rather we should think of it as a component of another screen (as it is being used in `PreviewScreen`). The [best practices](https://developer.android.com/topic/libraries/architecture/viewmodel#best-practices) for ViewModels state:

> Because of [their scoping](https://developer.android.com/topic/libraries/architecture/viewmodel#lifecycle), use ViewModels as implementation details of a screen level state holder. Don't use them as state holders of reusable UI components such as chip groups or forms. Otherwise, you'd get the same ViewModel instance in different usages of the same UI component under the same ViewModelStoreOwner.

Reply:
I have changed it to ViewModelScoped, but with my current changes, even that is not required I think. But that's mainly because of how ViewModel + Hilt doesn't support injecting the `viewModelScope` and thus leading to create `ScreenFlash` as direct dependency.

If the dependency graph becomes more complex later, I guess it'll be better to inject `ScreenFlash` as well with the `viewModelScope` passed to method parameter wherever needed. So I kept `CameraXCameraUseCase` to `ViewModelScoped` which makes more sense in general too, IMO.

Reply:
SGTM